frontend/server/src/main/java/org/pytorch/serve/wlm/Model.java

```diff
@@ -245,6 +247,90 @@ public void addFirst(Job job) {
         jobsDb.get(DEFAULT_DATA_QUEUE).addFirst(job);
     }
 
+    public boolean pollMgmtJob(String threadId, long waitTime, Map<String, Job> jobsRepo)
+            throws InterruptedException {
+        if (jobsRepo == null || threadId == null || threadId.isEmpty()) {
+            throw new IllegalArgumentException("Invalid input given provided");
+        }
+
+        // if (!jobsRepo.isEmpty()) {
+        //     throw new IllegalArgumentException(
+        //             "The jobs repo provided contains stale jobs. Clear them!!");
+        // }
+
+        LinkedBlockingDeque<Job> jobsQueue = jobsDb.get(threadId);
+        if (jobsQueue != null && !jobsQueue.isEmpty()) {
+            Job j = jobsQueue.poll(waitTime, TimeUnit.MILLISECONDS);
+            if (j != null) {
+                jobsRepo.put(j.getJobId(), j);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public void pollInferJob(Map<String, Job> jobsRepo, long waitTime, int batchSize) throws InterruptedException {
+        if(batchSize == 0)
+            return;
+
+        LinkedBlockingDeque<Job> jobsQueue;
+        try {
+            if (isUseJobTicket()) {
+                incNumJobTickets();
+            }
+            lock.lockInterruptibly();
+
+            jobsQueue = jobsDb.get(DEFAULT_DATA_QUEUE);
+
+            Job j = jobsQueue.poll(waitTime, TimeUnit.MILLISECONDS);
```

Comment:
This will cause WorkerThead run with empty job, waste CPU resource if there are no incoming requests.

Reply:
see above